lib/internal/modules/esm/loader.js

```diff
@@ -56,16 +58,32 @@ class Loader {
       await this._resolve(specifier, parentURL, defaultResolve);
```

Comment:
It might be useful to validate that the return type is actually an object. Otherwise the following error messages would still be weird (e.g., in case `5` is returned the error message would be about the `url` property).

Reply:
Sure, I didn't realise destructuring didn't throw for such cases actually.